packages/debugger/app/app.tsx

```diff
@@ -0,0 +1,187 @@
+"use client";
+/** requires client because signer is stored in local storage */
+
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { ConnectButton } from "@rainbow-me/rainbowkit";
+import { sendTransaction, switchChain } from "@wagmi/core";
+import {
+  FrameUI,
+  fallbackFrameContext,
+  OnTransactionFunc,
+} from "frames.js/render";
+import { FrameImageNext } from "frames.js/render/next";
+import { useFrame } from "frames.js/render/use-frame";
+import dynamic from "next/dynamic";
+import { useRouter } from "next/navigation";
+import { useCallback, useEffect, useState } from "react";
+import { useChainId, useConfig } from "wagmi";
+import { FrameDebugger } from "./components/frame-debugger";
+import { useFarcasterIdentity } from "./hooks/use-farcaster-identity";
+import { MockHubActionContext } from "./utils/mock-hub-utils";
+const LoginWindow = dynamic(() => import("./components/create-signer"), {
+  ssr: false,
+});
+
+export default function App({
+  searchParams,
+}: {
+  searchParams: Record<string, string>;
+}): JSX.Element {
+  const router = useRouter();
+  const url = searchParams.url ?? null;
+  const [urlInput, setUrlInput] = useState(
+    url || process.env.NEXT_PUBLIC_HOST || "http://localhost:3000"
+  );
+  const [mockHubContext, setMockHubContext] = useState<
+    Partial<MockHubActionContext>
+  >({
+    enabled: true,
+    requesterFollowsCaster: false,
+    casterFollowsRequester: false,
+    likedCast: false,
+    recastedCast: false,
+  });
+  const currentChainId = useChainId();
+
+  useEffect(() => {
+    if (url !== urlInput && url) {
+      setUrlInput(url);
+    }
+  }, [url]);
+  const signerState = useFarcasterIdentity();
+  const globalConfig = useConfig();
+
+  const onTransaction: OnTransactionFunc = useCallback(
+    async ({ transactionData }) => {
+      const { params, chainId, method } = transactionData;
+      if (!chainId.startsWith("eip155:")) {
+        alert(`debugger: Unrecognized chainId ${chainId}`);
+      }
+
+      const requestedChainId = parseInt(chainId.split("eip155:")[1]!);
+
+      const config = globalConfig;
+      if (currentChainId !== requestedChainId) {
+        await switchChain(config, {
+          chainId: requestedChainId,
+        });
+      }
+
+      try {
+        // Send the transaction
+        const transactionId = await sendTransaction(config, {
+          to: params.to,
+          data: params.data,
+          value: BigInt(params.value),
+        });
+        return transactionId;
+      } catch (error) {
+        console.error(error);
+        return null;
+      }
+    },
+    []
```

Comment:
think we should enable rule of hooks lint rule so it needs to be explicitly ignored for each param

Reply:
In my opinion it is better to use `useRef` to store the value and not use ignore rules since those can lead to bugs that are hard to find since you are ignoring the eslint warning completely.

So I think it is better to enable the hook dependencies rule and if you want to ignore some dependency, use ref for it. Much cleaner solution even if that means that there are 2 more lines of code.